src/commands/deploy/deploy.ts

```diff
@@ -158,7 +158,8 @@ export async function deploy(context: IActionContext, confirmDeployment: boolean
         deployContext.telemetry.properties.cancelStep = '';
```

Comment:
I think another root cause is disabling `max-func-body-length cyclomatic-complexity` in `deploy` because it makes it hard to follow all of the possible code-paths. Would like to see that fixed sometime soon, but not in this PR

Reply:
I'll assign this to @wwlorey 😁 

jk, but yes, I agree.  I'll start refactoring this after the release.  This code is getting to be a mess: https://github.com/microsoft/vscode-azureappservice/issues/1112

Reply:
As an example, I think you could make `getDefaultWebAppToDeploy` broader to just be `getWebAppToDeploy` similar to how I had `getDeployFsPath`. In other words it takes in whatever the target is and returns a web app after checking all the possible settings, node picker, target, etc.